x/tx/signing/aminojson/encoder.go

```diff
@@ -51,7 +51,12 @@ func cosmosDecEncoder(_ *Encoder, v protoreflect.Value, w io.Writer) error {
 		if val == "" {
 			return jsonMarshal(w, "0")
 		}
-		return jsonMarshal(w, val)
+		var dec math.LegacyDec
+		err := dec.Unmarshal([]byte(val))
```

Comment:
this seems breaks json encoding of grpc response, `dec.Unmarshal` don't handle decimal point strings.

Reply:
I'm trying to reproduce what you're claiming @yihuang.. from this branch I applied the patch below.

```diff
diff --git a/simapp/go.mod b/simapp/go.mod
index 3c7157da8d..1506231eab 100644
--- a/simapp/go.mod
+++ b/simapp/go.mod
@@ -205,6 +205,8 @@ require (
 //     <temporary replace>
 // )

+replace cosmossdk.io/x/tx => ../x/tx
+
 // Below are the long-lived replace of the SimApp
 replace (
        // use cosmos fork of keyring
```

The JSON response look OK by way of these commands.  How can I reproduce what you're saying?

```
❯ go run simd/main.go start &
❯ curl -s curl http://localhost:1317/cosmos/distribution/v1beta1/community_pool
{"pool":[{"denom":"stake","amount":"1357.100000000000000000"}]}⏎
```
```

Reply:
Oh I see now, we need to backport this as well.

https://github.com/cosmos/cosmos-sdk/pull/22161

Reply:
cherry-picking this in here.